webroot/js/toolbar.js

```diff
@@ -97,6 +97,13 @@ if (elem) {
     doc.addEventListener('DOMContentLoaded', onReady, false);
     doc.addEventListener('DOMContentLoaded', proxyAjaxOpen, false);
     doc.addEventListener('DOMContentLoaded', proxyAjaxSend, false);
+
+    if (!window.__debugKitTurbolinksListenerApplied) {
```

Comment:
This checks if the debugkit listeners are already applied

Reply:
When would this block be re-run?

Reply:
Literally on every page navigation. The result of that is that the same listener (e.g. `onReady`) doesn't get added multiple times. Otherwise the listener will be added multiple times and the e.g. `onReady`-method gets called <page navigations done>^2 times which can cause serious bugs.

The check could also wrap the `DOMContentLoaded` listeners, but its not required. It would "just" improve performance.